src/librustdoc/html/render.rs

```diff
@@ -490,7 +495,12 @@ pub fn run(mut krate: clean::Crate,
            sort_modules_alphabetically: bool,
            themes: Vec<PathBuf>,
            enable_minification: bool,
-           id_map: IdMap) -> Result<(), Error> {
+           id_map: IdMap,
+           disable_index_page: bool,
+           index_page: Option<PathBuf>,
+           matches: &getopts::Matches,
+           diag: &errors::Handler,
```

Comment:
I think i want to refactor a lot of how we handle configuration and CLI arguments, to make a central Config struct instead of trying to pass the Matches struct around everywhere. (Or avoiding passing it everywhere and instead getting a function with over a dozen arguments, like we've reached with `core::run`.)

Reply:
I think it'll be better in another PR, otherwise we'll have a lot of noise not related to this PR. But yes, I strongly agree with you.